src/Workspaces/CSharp/Portable/Editing/CSharpImportAdder.cs

```diff
@@ -142,91 +129,137 @@ private void AddImportedMembers(ImmutableArray<INamespaceSymbol> namespaceSymbol
+                produceItems: static (node, onItemsFound, args, cancellationToken) =>
+                {
+                    var (self, containsAnonymousMethods, _) = args;
+                    if (node is SimpleNameSyntax nameSyntaxNode)
+                        self.ProduceConflicts(nameSyntaxNode, onItemsFound, cancellationToken);
+                    else if (node is MemberAccessExpressionSyntax memberAccessExpressionNode)
+                        self.ProduceConflicts(memberAccessExpressionNode, containsAnonymousMethods, onItemsFound, cancellationToken);
+                    else
+                        throw ExceptionUtilities.Unreachable();
+
+                    return Task.CompletedTask;
+                },
+                consumeItems: static async (items, args, cancellationToken) =>
+                {
+                    var (_, _, conflicts) = args;
+                    await foreach (var conflict in items)
+                        conflicts.Add(conflict);
+                },
+                args: (self: this, containsAnonymousMethods, conflicts),
+                cancellationToken).ConfigureAwait(false);
         }
 
-        public override void VisitAnonymousMethodExpression(AnonymousMethodExpressionSyntax node)
+        private void CollectInfoFromContainer(SyntaxNode container, ArrayBuilder<SyntaxNode> nodes, out bool containsAnonymousMethods)
         {
-            var previousInAnonymousMethod = _inAnonymousMethod;
-            _inAnonymousMethod = true;
-            base.VisitAnonymousMethodExpression(node);
-            _inAnonymousMethod = previousInAnonymousMethod;
+            containsAnonymousMethods = false;
+
+            foreach (var node in container.DescendantNodesAndSelf())
+            {
+                switch (node.Kind())
+                {
+                    case SyntaxKind.IdentifierName:
+                    case SyntaxKind.GenericName:
+                        if (IsPotentialConflictWithImportedType((SimpleNameSyntax)node))
+                            nodes.Add(node);
+                        break;
+                    case SyntaxKind.SimpleMemberAccessExpression:
+                    case SyntaxKind.PointerMemberAccessExpression:
+                        if (IsPotentialConflictWithImportedExtensionMethod((MemberAccessExpressionSyntax)node))
+                            nodes.Add(node);
+                        break;
+                    case SyntaxKind.SimpleLambdaExpression:
+                    case SyntaxKind.ParenthesizedLambdaExpression:
+                    case SyntaxKind.AnonymousMethodExpression:
+                        // Track if we've seen an anonymous method or not.  If so, because of how the language binds lambdas and
+                        // overloads, we'll assume any method access we see inside (instance or otherwise) could end up conflicting
+                        // with an extension method we might pull in.
+                        containsAnonymousMethods = true;
```

Comment:
ok to preserve this.  but this genuinely makes no sense wrt to how the conflict checker has to work.